pkgs/development/python-modules/imgaug/default.nix

```
@@ -1,30 +1,54 @@
-{ stdenv, buildPythonPackage, fetchPypi, numpy, scipy, scikitimage, opencv3, six }:
+{ stdenv
+,buildPythonPackage
+,fetchurl
+,numpy
+,scipy
+,scikitimage
+,opencv3
+,six
+,imageio
+,shapely
+,pytest
+ }:
 
 buildPythonPackage rec {
   pname = "imgaug";
   version = "0.3.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "e1354d41921f1b306b50c5141b4870f17e81b531cae2f5c3093da9dc4dcb3cf4";
+  src = fetchurl {
+    url = "https://github.com/aleju/imgaug/archive/c3d99a420efc45652a1264920dc20378a54b1325.zip";
+    sha256 = "sha256:174nvhyhdn3vz0i34rqmkn26840j3mnfr55cvv5bdf9l4y9bbjq2";
   };
 
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace "opencv-python-headless" ""
+    substituteInPlace setup.py \
+      --replace "opencv-python-headless" "" 
+    substituteInPlace pytest.ini \
+      --replace "--xdoctest --xdoctest-global-exec=\"import imgaug as ia\nfrom imgaug import augmenters as iaa\"" ""
+  '';
+
   propagatedBuildInputs = [
     numpy
     scipy
     scikitimage
     opencv3
     six
+    imageio
+    shapely
   ];
 
-  # disable tests when there are no tests in the PyPI archive
-  doCheck = false;
+  checkPhase = ''
+     python -m pytest ./test
+  '';
+
+  checkInputs = [ opencv3 pytest ];
 
   meta = with stdenv.lib; {
     homepage = https://github.com/aleju/imgaug;
     description = "Image augmentation for machine learning experiments";
     license = licenses.mit;
-    maintainers = with maintainers; [ cmcdragonkai ];
-    broken = true; # opencv-python bindings aren't available yet, and look non-trivial
+    maintainers = with maintainers; [ cmcdragonkai rakesh4g ];
```

```suggestion
    maintainers = with maintainers; [ cmcdragonkai rakesh4g ];
    platforms = platforms.linux;
```
looks like darwin has odd behavior:
```
Current thread 0x0000700003d9d000 (most recent call first):
  File "/private/tmp/nix-build-python3.7-imgaug-0.3.0.drv-0/imgaug-c3d99a420efc45652a1264920dc20378a54b1325/imgaug/augmenters/blur.py", line 262 in blur_gaussian_
  File "/private/tmp/nix-build-python3.7-imgaug-0.3.0.drv-0/imgaug-c3d99a420efc45652a1264920dc20378a54b1325/imgaug/augmenters/blur.py", line 466 in _augment_batch
  File "/private/tmp/nix-build-python3.7-imgaug-0.3.0.drv-0/imgaug-c3d99a420efc45652a1264920dc20378a54b1325/imgaug/augmenters/meta.py", line 606 in augment_batch
  File "/private/tmp/nix-build-python3.7-imgaug-0.3.0.drv-0/imgaug-c3d99a420efc45652a1264920dc20378a54b1325/imgaug/multicore.py", line 506 in _Pool_worker
  File "/private/tmp/nix-build-python3.7-imgaug-0.3.0.drv-0/imgaug-c3d99Fatal Python error: Segmentation fault
```

---

Done.